kernel: bump 6.12 to 6.12.45

graysky2 opened the pull request:
Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.45

Removed upstreamed:
  generic/backport-6.12/626-v6.17-net-ipv4-fix-regression-in-local-broadcast-routes.patch[1]
  mediatek/patches-6.12/051-v6.17-thermal-drivers-mediatek-lvts_thermal-Change-lvts-co.patch[2]
  mediatek/patches-6.12/052-v6.17-thermal-drivers-mediatek-lvts_thermal-Add-lvts-comma.patch[3]
  mediatek/patches-6.12/053-v6.17-thermal-drivers-mediatek-lvts_thermal-Add-mt7988-lvt.patch[4]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.45&id=81ff76c1b08827bc81779400a3640f102a9a9ade
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.45&id=d1f4b09d9bb991c0fe039511520c6e59f1b42ec1
3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.45&id=739229eb4d5cd009d81ad8946fdd4bb5ec790c2e
4. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.45&id=9a7141d4808dcb833f87154af88560c785306cd2

Build system: x86/64
Build-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64-glibc
Run-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64-glibc

graysky2 committed:
kernel: bump 6.12 to 6.12.45

Changelog: https://cdn.kernel.org/pub/linux/kernel/v6.x/ChangeLog-6.12.45

Removed upstreamed:
  generic/backport-6.12/626-v6.17-net-ipv4-fix-regression-in-local-broadcast-routes.patch[1]
  mediatek/patches-6.12/051-v6.17-thermal-drivers-mediatek-lvts_thermal-Change-lvts-co.patch[2]
  mediatek/patches-6.12/052-v6.17-thermal-drivers-mediatek-lvts_thermal-Add-lvts-comma.patch[3]
  mediatek/patches-6.12/053-v6.17-thermal-drivers-mediatek-lvts_thermal-Add-mt7988-lvt.patch[4]

All other patches automatically rebased.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.45&id=81ff76c1b08827bc81779400a3640f102a9a9ade
2. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.45&id=d1f4b09d9bb991c0fe039511520c6e59f1b42ec1
3. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.45&id=739229eb4d5cd009d81ad8946fdd4bb5ec790c2e
4. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=v6.12.45&id=9a7141d4808dcb833f87154af88560c785306cd2

Build system: x86/64
Build-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64-glibc
Run-tested: flogic/xiaomi_redmi-router-ax6000-ubootmod, ramips/tplink_archer-a6-v3, x86/64-glibc

Signed-off-by: John Audia <therealgraysky@proton.me>
Link: https://github.com/openwrt/openwrt/pull/19956
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

schuettecarsten commented:
Works on bcm27xx/bcm2711 (rpi4)

DragonBluep commented on target/linux/generic/backport-6.6/626-v6.17-net-ipv4-fix-regression-in-local-broadcast-routes.patch:
```diff
@@ -1,47 +0,0 @@
-From 5189446ba995556eaa3755a6e875bc06675b88bd Mon Sep 17 00:00:00 2001
```
This patch belongs to the 6.6 kernel patch set. It should not be removed.

hauke commented:
This does not compile on mips malat:
```
../mt76-2025.09.04~6c902ccb/mt7996/mac.c: In function 'mt7996_rro_rx_process':
../mt76-2025.09.04~6c902ccb/mt7996/mac.c:1851:26: error: left shift count >= width of type [-Werror=shift-count-overflow]
 1851 |                 dma_addr <<= 32;
      |                          ^~~
../mt76-2025.09.04~6c902ccb/mt7996/mac.c:1937:42: error: left shift count >= width of type [-Werror=shift-count-overflow]
 1937 |                                 dma_addr <<= 32;
      |                                          ^~~
cc1: all warnings being treated as errors
```

hauke commented:
@abajk Some lantiq dts files still use  `etop`, do we have to adapt this?
```
$ git grep etop target/linux/lantiq/files/
target/linux/lantiq/files/arch/mips/boot/dts/lantiq/amazonse.dtsi:              gsw: etop@e180000 {
target/linux/lantiq/files/arch/mips/boot/dts/lantiq/amazonse.dtsi:                      compatible = "lantiq,etop-xway";
target/linux/lantiq/files/arch/mips/boot/dts/lantiq/ar9.dtsi:           gsw: etop@e180000 {
target/linux/lantiq/files/arch/mips/boot/dts/lantiq/ar9.dtsi:                   compatible = "lantiq,etop-xway";
target/linux/lantiq/files/arch/mips/boot/dts/lantiq/danube.dtsi:                gsw: etop@e180000 {
target/linux/lantiq/files/arch/mips/boot/dts/lantiq/danube.dtsi:                        compatible = "lantiq,etop-xway";
```
See:  https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.12.y&id=8c431ea8f3f795c4b9cfa57a85bc4166b9cce0ac

abajk commented:
> @abajk Some lantiq dts files still use `etop`, do we have to adapt this?

Yes. The name of the etop node must be changed to ethernet.

graysky2 committed:
lantiq: adapt upstream commit to some dts files

Some lantiq dts files still use etop but 6.12.45[1] changed to ethernet
so bring them into parity with this change.

1. https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/commit/?h=linux-6.12.y&id=8c431ea8f3f795c4b9cfa57a85bc4166b9cce0ac

Co-Authored-by: Hauke Mehrtens <hauke@hauke-m.de>
Co-Authored-by: Aleksander Bajkowski <olek2@wp.pl>
Signed-off-by: John Audia <therealgraysky@proton.me>
Link: https://github.com/openwrt/openwrt/pull/19956
Signed-off-by: Hauke Mehrtens <hauke@hauke-m.de>

graysky2 commented:
> Some lantiq dts files still use etop, do we have to adapt this?
@hauke - I make this change and add you to the `Co-Authored-By` line. If you object please speak up now.

GithPK commented:
Arcadyan AW1000 does not work properly with the last two compilations: 
SNAPSHOT (r30963-79d3db7447)
SNAPSHOT (r30972-0ee0846be1)
Modem virtual DHCP interface does not start properly and there is no internet connection available.

Works properly with previous SNAPSHOT r30910-886382ba25

GithPK commented:
> > Arcadyan AW1000 does not work properly with the last two compilations: SNAPSHOT (r30963-79d3db7447) SNAPSHOT (r30972-0ee0846be1) Modem virtual DHCP interface does not start properly and there is no internet connection available.
> > Works properly with previous SNAPSHOT r30910-886382ba25
> 
> Could you please open a separate issue, as the versions you mention have nothing to do with this commit?

I though the new kernel version has something to do with it.
